From 18fa782bec4763b92e68a69a0445859feb405287 Mon Sep 17 00:00:00 2001
From: StanleyYP Wang <StanleyYP.Wang@mediatek.com>
Date: Thu, 15 Feb 2024 14:30:02 +0800
Subject: [PATCH 049/137] mtk: mac80211: fix ieee80211_ht_cap_ie_to_sta_ht_cap
 warn on

Fix ieee80211_ht_cap_ie_to_sta_ht_cap warning.
For MLD with a 2/5G primary link, auth/assoc is done in the 2G or 5G link.
Therefore, 6G link will enter ieee80211_ht_cap_ie_to_sta_ht_cap, as elems->ht_cap_elem of 2/5G is NOT NULL.
This should be avoided; otherwise, if 6G is bw 320, then the warning will be triggered.

Signed-off-by: StanleyYP Wang <StanleyYP.Wang@mediatek.com>
---
 net/mac80211/mlme.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)


--- a/net/mac80211/mlme.c
+++ b/net/mac80211/mlme.c
@@ -6058,7 +6058,8 @@
 	sband = local->hw.wiphy->bands[link->conf->chanreq.oper.chan->band];
 
 	/* Set up internal HT/VHT capabilities */
-	if (elems->ht_cap_elem && link->u.mgd.conn.mode >= IEEE80211_CONN_MODE_HT)
+	if (elems->ht_cap_elem && link->u.mgd.conn.mode >= IEEE80211_CONN_MODE_HT &&
+	    !is_6ghz)
 		ieee80211_ht_cap_ie_to_sta_ht_cap(sdata, &sband->ht_cap,
 						  elems->ht_cap_elem,
 						  link_sta);
